api/types/webauthn/webauthn.proto

```diff
@@ -0,0 +1,257 @@
+syntax = "proto3";
+
+// Package WebAuthn maps WebAuthn messages to protocol buffers.
+//
+// The mapping is designed to match both the WebAuthn specification and the
+// capabilities of current browser implementations.
+//
+// REST-based Teleport APIs will make an effort to transmit or embed JSON
+// messages matching the github.com/duo-labs/webauthn reference implementation,
+// to allow for easy browser integration. gRPC APIs are not meant for REST use
+// and thus make no such promises, although the correspondence should be
+// obvious.
+//
+// Note that, ordinarily, various fields in WebAuthn messages are encoded using
+// "RawURLEncoding" (aka, base64 URL encoding without padding). This is not the
+// case for _any_ of the fields mapped here, all bytes fields are transmitted
+// raw/unencoded.
+package webauthn;
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// -----------------------------------------------------------------------------
+// WebAuthn messages used by server storage.
+// -----------------------------------------------------------------------------
+
+// SessionData stored by the Relying Party during authentication ceremonies.
+// Mirrors https://pkg.go.dev/github.com/duo-labs/webauthn/webauthn#SessionData.
+message SessionData {
+    // Raw challenge used for the ceremony.
+    bytes challenge = 1 [ (gogoproto.jsontag) = "challenge,omitempty" ];
+    // Raw User ID.
+    bytes user_id = 2 [ (gogoproto.jsontag) = "userId,omitempty" ];
+    // Raw Credential IDs of the credentials allowed for the ceremony.
+    repeated bytes allow_credentials = 3 [ (gogoproto.jsontag) = "allowCredentials,omitempty" ];
+    // Requested user verification.
+    // https://www.w3.org/TR/webauthn-2/#enum-userVerificationRequirement.
+    string user_verification = 4 [ (gogoproto.jsontag) = "userVerification,omitempty" ];
+}
+
+// -----------------------------------------------------------------------------
+// Assertion (aka login).
+// -----------------------------------------------------------------------------
+
+// Credential assertion used for login ceremonies.
+message CredentialAssertion { PublicKeyCredentialRequestOptions public_key = 1; }
```

Comment:
does this need to be a nested field?

Reply:
I'd rather play it safe and mimic browser APIs, in case they end up adding something other than `public_key` to the top-level message.